src/legacy/core_plugins/data/public/query/query_bar/components/query_bar_input.tsx

```diff
@@ -468,11 +468,22 @@ export class QueryBarInputUI extends Component<Props, State> {
       <EuiOutsideClickDetector onOutsideClick={this.onOutsideClick}>
         <div
           style={{ position: 'relative' }}
-          role="combobox"
+          {...{ role: 'combobox' }} // using awkward syntax to avoid triggering false jsx-a11y/role-has-required-aria-props linting failure
```

Comment:
I think we shouldn't do it in that way. Either if we are happy with the implementation as it is, we should just use eslint-disable comments above those line to disable that check and explain why it's not needed here.

To be honest I think we should rather actually fix this implementation to comply with the ARIA specification, meaning there always must be an `aria-expanded` and `aria-controls` on an `role="combobox"` element.

`aria-expanded` already is no problem, so it's only `aria-controls`. Two suggestions:

a) we could potentially just leave an empty `aria-controls` in there, not pointing to anything (not sure compliance wise how much more standard that would be).
b) pull the `kbnTypeahead__items` a level higher so it basically sits on the "wrapping" element, that would always be visible and not hidden (and if there is no such element, create one).

Reply:
My original goal was to disable eslint. I fiddled with eslint disable comments for a long time here and I couldn't get something working where I didn't end up disabling checking on huge swathes of code. Maybe you have a suggestion here?

Directly addressing your two other suggestions: 
a) Leaving an empty `aria-controls` will be flagged as an issue (it's not practically an issue but tools will flag it so 3rd party auditors will flag so we'll get bothered about it). 
b) My goal with this PR is to do the minimal work for maximum bang so fixing the typeahead is out of scope for this PR. Fixing one thing always snowballs into another so I don't even want to poke it. (Also, with our many floating typeahead implementations, fixing the one not in EUI seems like throwaway work.)

Reply:
I rolled this spread in with the other `aria-*` attributed being spread into this element so it looks less weird now. Thoughts? 

Preview:
```jsx
const isSuggestionsVisible = this.state.isSuggestionsVisible && {
  'aria-controls': 'kbnTypeahead__items',
  'aria-owns': 'kbnTypeahead__items',
};
const airaCombobox = Object.assign({ role: 'combobox' }, isSuggestionsVisible);

return (<div {...ariaCombobox}></div>);
```